### PR TITLE
Fix(Curriculum): Constructor function not explained before ES6 #17865

### LINKED ISSUE #17865: Constructor function not explained before ES6
#### Describe your problem and - if possible - how to reproduce it
'ES6: Use class Syntax to Define a Constructor Function' talks about class syntax without we knowing anything about constructor functions, but I think, it's important to know about constructor functions first.

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function

### PR DESCRIPTION
**Pull request by hilliarj**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #17865

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md]:
```suggestion
2. The constructor method is a special method for creating and initializing an object created with a class. You will learn more about it in the Object Oriented Programming section of the Javascript Algorithms And Data Structures Certification
```
This is a better wording since if anyone is going through the curriculum, they won't have to hop on to another challenge from in between somewhere else

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Have a look at the comment

**Comment by hilliarj** [comment]
Updated the PR. Thanks for the suggestion!

**Review by thecodingaviator — APPROVED** [review]
LGTM! 👍

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md]: Per discussions on issue https://github.com/freeCodeCamp/freeCodeCamp/issues/35464, this should be:
```suggestion
<strong>Notes:</strong><br>
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md]: Per same discussion on issue, this should be a unordered list item:
```suggestion
* UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md]: Per same discussion on issue, this should be a unordered list item:
```suggestion
* The constructor method is a special method for creating and initializing an object created with a class. You will learn more about it in the Object Oriented Programming section of the Javascript Algorithms And Data Structures Certification
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my suggestions.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md]: You can not use GitHub markdown for the Description or Instruction sections.  You must use an unordered list like:
```
<ul>
  <li>...</li>
  <li>..</li>
</ul>
```
Make sure you are testing all of your changes locally so you can see how it will actually render on the site.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md]: Same issue as pointed out above.

**Review by RandellDawson — CHANGES_REQUESTED** [review]
You can not use GitHub markdown.  Please test any non-content changes like HTML, CSS, or JavaScript locally first, before creating PR.

**Comment by hilliarj** [comment]
Sorry for the delay, been busy with tests at college and family affairs.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
LGTM.  Just waiting on #34538 to be reviewed/merged before merging this one.

**Review by cmccormack — APPROVED** [review]
This looks good to me 👍.  Thank you for your contribution!

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**
